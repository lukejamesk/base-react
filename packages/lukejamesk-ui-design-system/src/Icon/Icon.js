import React from 'react'
import PropTypes from 'prop-types'
import clsx from 'clsx'
import { Icon as IconCore } from '@lukejamesk/core'

const variantMappiping = {
  light: 'ljk-text-white',
  dark: 'ljk-text-black',
  primary: 'ljk-text-primary',
  secondary: 'ljk-text-secondary',
}

const sizeMapping = {
  s: 'ljk-text-sm',
  m: 'ljk-text-base',
  l: 'ljk-text-4xl',
  auto: '',
}

export const iconVariants = ['dark', 'light', 'primary', 'secondary']
export const iconSizes = ['s', 'm', 'l']

const Icon = ({ name, variant = 'dark', size = 'auto' }) => {
  return (
    <span className={clsx(variantMappiping[variant], sizeMapping[size])}>
      <IconCore name={name} className="ljk-fill-current" />
    </span>
  )
}

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  variant: PropTypes.oneOf(iconVariants),
  size: PropTypes.oneOf(iconSizes),
}

export default Icon
