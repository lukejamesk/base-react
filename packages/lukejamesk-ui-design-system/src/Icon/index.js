export { default, iconVariants, iconSizes } from './Icon'
