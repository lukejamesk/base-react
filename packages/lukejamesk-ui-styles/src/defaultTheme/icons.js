const icons = {
  s: 12,
  m: 24,
  l: 48,
}

export default icons
