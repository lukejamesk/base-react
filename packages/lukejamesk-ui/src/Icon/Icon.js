import React from 'react'
import PropTypes from 'prop-types'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

const Icon = ({ name, className }) => {
  return <FontAwesomeIcon icon={name} className={className} />
}

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
}

export default Icon
