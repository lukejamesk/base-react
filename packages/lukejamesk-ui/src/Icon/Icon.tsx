import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { IconProp } from '@fortawesome/fontawesome-svg-core'

export type IconName = IconProp

export type IconProps = {
  name: IconName
  className?: string
}

const Icon: React.FC<IconProps> = ({ name, className }) => {
  return <FontAwesomeIcon icon={name} className={className} />
}

export default Icon
