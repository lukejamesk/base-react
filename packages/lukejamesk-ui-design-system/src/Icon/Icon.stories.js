import React from 'react'
import { text, select } from '@storybook/addon-knobs'
import { forEach, compose, sort, comparator, lt, keys, map } from 'ramda'
import * as AllIcons from '@lukejamesk/icons'

import { library } from '@lukejamesk/core'
import Icon, { iconSizes, iconVariants } from './Icon'

const getKeys = compose(sort(comparator(lt)), keys)

forEach((icon) => {
  library.add(AllIcons[icon])
}, getKeys(AllIcons))

export default {
  title: 'Design System/Icon',
  component: Icon,
}

export const Default = () => (
  <Icon
    name={text('icon', 'arrow-down')}
    variant={select('Variant', iconVariants, iconVariants[0])}
    size={select('Size', iconSizes, iconSizes[0])}
  />
)

export const All = () => (
  <div className="ljk-grid ljk-grid-cols-12 ljk-gap-1">
    {map(
      (icon) => (
        <Icon
          name={AllIcons[icon].iconName}
          variant={select('Variant', iconVariants, iconVariants[0])}
          size={select('Size', iconSizes, iconSizes[0])}
        />
      ),
      getKeys(AllIcons),
    )}
  </div>
)
