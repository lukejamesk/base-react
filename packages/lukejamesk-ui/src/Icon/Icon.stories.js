import React from 'react'
import { text } from '@storybook/addon-knobs'
import { coffee } from '@lukejamesk/icons'
import library from '../library'
import Icon from './Icon'

library.add(coffee)

export default {
  title: 'UI|Icon',
}

export const Default = () => <Icon name={text('icon', 'coffee')} variant="primary" />
