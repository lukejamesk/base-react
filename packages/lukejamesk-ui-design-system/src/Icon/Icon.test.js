import React from 'react'
import { render } from 'enzyme'
import { library } from '@lukejamesk/core'
import { coffee } from '@lukejamesk/icons'
import { forEach } from 'ramda'
import Icon, { iconSizes, iconVariants } from './Icon'

library.add(coffee)

describe('<ButtonIcon />', () => {
  forEach((size) => {
    forEach((variant) => {
      describe(`Size: ${size} and Variant: ${variant}`, () => {
        it('should render', () => {
          const wrapper = render(<Icon size={size} variant={variant} name="coffee" />)
          expect(wrapper).toMatchSnapshot()
        })
      })
    }, iconVariants)
  }, iconSizes)
})
