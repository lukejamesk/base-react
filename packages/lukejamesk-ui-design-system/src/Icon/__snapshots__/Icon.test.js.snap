// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonIcon /> Size: l and Variant: dark should render 1`] = `
<span
  class="ljk-text-black ljk-text-4xl"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-coffee fa-w-20 ljk-fill-current"
    data-icon="coffee"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
      fill="currentColor"
    />
  </svg>
</span>
`;

exports[`<ButtonIcon /> Size: l and Variant: light should render 1`] = `
<span
  class="ljk-text-white ljk-text-4xl"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-coffee fa-w-20 ljk-fill-current"
    data-icon="coffee"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
      fill="currentColor"
    />
  </svg>
</span>
`;

exports[`<ButtonIcon /> Size: l and Variant: primary should render 1`] = `
<span
  class="ljk-text-primary ljk-text-4xl"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-coffee fa-w-20 ljk-fill-current"
    data-icon="coffee"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
      fill="currentColor"
    />
  </svg>
</span>
`;

exports[`<ButtonIcon /> Size: l and Variant: secondary should render 1`] = `
<span
  class="ljk-text-secondary ljk-text-4xl"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-coffee fa-w-20 ljk-fill-current"
    data-icon="coffee"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
      fill="currentColor"
    />
  </svg>
</span>
`;

exports[`<ButtonIcon /> Size: m and Variant: dark should render 1`] = `
<span
  class="ljk-text-black ljk-text-base"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-coffee fa-w-20 ljk-fill-current"
    data-icon="coffee"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
      fill="currentColor"
    />
  </svg>
</span>
`;

exports[`<ButtonIcon /> Size: m and Variant: light should render 1`] = `
<span
  class="ljk-text-white ljk-text-base"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-coffee fa-w-20 ljk-fill-current"
    data-icon="coffee"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
      fill="currentColor"
    />
  </svg>
</span>
`;

exports[`<ButtonIcon /> Size: m and Variant: primary should render 1`] = `
<span
  class="ljk-text-primary ljk-text-base"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-coffee fa-w-20 ljk-fill-current"
    data-icon="coffee"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
      fill="currentColor"
    />
  </svg>
</span>
`;

exports[`<ButtonIcon /> Size: m and Variant: secondary should render 1`] = `
<span
  class="ljk-text-secondary ljk-text-base"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-coffee fa-w-20 ljk-fill-current"
    data-icon="coffee"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
      fill="currentColor"
    />
  </svg>
</span>
`;

exports[`<ButtonIcon /> Size: s and Variant: dark should render 1`] = `
<span
  class="ljk-text-black ljk-text-sm"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-coffee fa-w-20 ljk-fill-current"
    data-icon="coffee"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
      fill="currentColor"
    />
  </svg>
</span>
`;

exports[`<ButtonIcon /> Size: s and Variant: light should render 1`] = `
<span
  class="ljk-text-white ljk-text-sm"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-coffee fa-w-20 ljk-fill-current"
    data-icon="coffee"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
      fill="currentColor"
    />
  </svg>
</span>
`;

exports[`<ButtonIcon /> Size: s and Variant: primary should render 1`] = `
<span
  class="ljk-text-primary ljk-text-sm"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-coffee fa-w-20 ljk-fill-current"
    data-icon="coffee"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
      fill="currentColor"
    />
  </svg>
</span>
`;

exports[`<ButtonIcon /> Size: s and Variant: secondary should render 1`] = `
<span
  class="ljk-text-secondary ljk-text-sm"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-coffee fa-w-20 ljk-fill-current"
    data-icon="coffee"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
      fill="currentColor"
    />
  </svg>
</span>
`;
